ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -893,6 +893,7 @@ class EclCmdRun : public EclCmdWithEclTarget
             break;
         default:
             fprintf(stderr, "%s %s\n", respwuid.str(), resp->getState());
+            ret = 1;
```

Comment:
The specific issue was that when state is failed you are not returning an error code...

I would think you want a 0 return if completed or if compiled with action=compile, an return code of (say) 4 for "failed" and an error code of (say) 2 for other states that imply 
"still running..."